core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1809,7 +1809,9 @@ private[spark] class BlockManager(
     blocksToRemove.size
   }
 
-  def decommissionBlockManager(): Unit = synchronized {
+  def decommissionBlockManager(): Unit = storageEndpoint.ask(DecommissionBlockManager)
```

Comment:
Why?

Reply:
?

Reply:
Why did you make this change?

Reply:
If I unsterstand your question correctly:

We didn't really change the `decommissionBlockManager`. The original `decommissionBlockManager` has been renamed to `decommissionSelf` to avoid the naming collision.

Reply:
Makes sense, although maybe introducing a new name instead of changing the use of a previous function name would be easier for verifying.